sources/config.h

```diff
@@ -38,6 +38,8 @@ class Config : public QObject, ConfigInterface
     Q_INTERFACES(ConfigInterface)
 
 public:
+    Q_PROPERTY (bool             valid              READ isValid                                    CONSTANT)
```

Comment:
Missing NOTIFY. That is needed so we can connect to a signal from QML.

Reply:
Do you really need to expose all properties to QML?
I would rather remove as many properties as possible if they are not used in the UI at the moment. Configuration setting & validation happens only at app startup and in the YIO API (which is a cpp class). Or did I miss something?

Reply:
You might want to display an error notification on the remote. That's the reason I think we should have at least one signal.

Reply:
At the time the configuration is loaded there's no UI. For that we need something new and the error is already propagated all the way to the application loading in main.cpp.
For the websocket API there's a TODO for an error response. In my opinion it doesn't make sense to display an error on the remote when using the web-configurator:  
In case the web-configurator sends invalid config data, it is not stored in the config object. I.e. the old configuration is still active on the remote.
We should keep the config & jsonfile classes as lean as possible and only expose QML properties and invokable methods which are really required at the moment. This makes refactoring much easier.

Reply:
Right. Makes sense. Agree about the web configurator. The error should be on the web interface.

Reply:
Can I remove all unused properties or would you like to keep them?